src/access/AccessRestrictionProvider.cpp

```diff
@@ -197,13 +197,19 @@ CHIP_ERROR AccessRestrictionProvider::DoCheck(const std::vector<Entry> & entries
                 if (requestPath.requestType == RequestType::kAttributeReadRequest ||
                     requestPath.requestType == RequestType::kAttributeWriteRequest)
                 {
-                    return CHIP_ERROR_ACCESS_RESTRICTED_BY_ARL;
+                    if (!IsGlobalAttribute(requestPath.entityId.value()))
```

Comment:
Followup: `AreRestrictionsAllowed` should probably be changed to return false for the "global attribute" case (e.g. by passing in the entity id only when the requestType or restrictionType are attribute-related?)